opencti-platform/opencti-graphql/src/modules/support/support-domain.ts

```diff
@@ -0,0 +1,335 @@
+  }
+  return entityUpdated;
+};
+
+/**
+ * When all nodes have sent files to support folder, one node need to zip them all, upload zip to s3 and update the entity with path to zip.
+ * Or this can be called by the "force zip" request also.
+ * @param context
+ * @param user
+ * @param entity
+ */
+export const zipAllSupportFiles = async (context: AuthContext, user: AuthUser, entity: BasicStoreEntitySupportPackage) => {
+  const zipLocalRootFolder = join(SUPPORT_LOG_RELATIVE_LOCAL_DIR, entity.id);
+  const zipLocalFullFolder: string = join(zipLocalRootFolder, NODE_INSTANCE_ID);
+
+  if (!fs.existsSync(zipLocalFullFolder)) {
+    fs.mkdirSync(zipLocalFullFolder, { recursive: true });
+  }
+
+  await downloadAllLogFiles(user, `${entity.package_upload_dir}/`, zipLocalFullFolder);
+
+  const zipName = `${entity.id}.zip`;
+  const zipFullpath = join(SUPPORT_LOG_RELATIVE_LOCAL_DIR, zipName);
+  // FIXME I'm quite sure that we can generate a zip steam without writing on filesystem.
+  await archiveFolderToZip(zipLocalFullFolder, zipFullpath);
+
+  const updatedEntity2 = await uploadArchivedSupportPackageToS3(context, user, SUPPORT_LOG_RELATIVE_LOCAL_DIR, zipName, entity);
+
+  // Cleaning zip folder and zip file locally
+  if (cleanupFiles) {
+    if (fs.existsSync(zipLocalRootFolder)) {
+      fs.rmSync(zipLocalRootFolder, { recursive: true, force: true });
+    }
+
+    if (fs.existsSync(zipFullpath)) {
+      fs.rmSync(zipFullpath, { recursive: true, force: true });
+    }
+  }
+
+  const updateInput = [
+    { key: 'package_url', value: [updatedEntity2.package_url], operation: EditOperation.Replace },
+    { key: 'package_status', value: [PackageStatus.Ready], operation: EditOperation.Replace }
+  ];
+  await updateAttribute(context, user, entity.id, ENTITY_TYPE_SUPPORT_PACKAGE, updateInput);
+};
+
+const createAndUploadTelemetry = async (context: AuthContext, user: AuthUser, entity: StoreEntitySupportPackage) => {
+  const filename = `telemetry-${new Date().getTime()}.json`;
+  const newLocalFile = join(SUPPORT_LOG_RELATIVE_LOCAL_DIR, filename);
+  fs.writeFileSync(newLocalFile, '{ \'message\': \'Telemetry not implemented yet\'}', {});
+
+  const uploadDirectory = getS3UploadFolder(entity.id);
+  logApp.info(`Sending telemetry to S3 - send to ${uploadDirectory}.`);
+  await uploadToStorage(context, user, uploadDirectory, fileToReadStream(SUPPORT_LOG_RELATIVE_LOCAL_DIR, filename, filename, 'application/json'), {});
+
+  if (cleanupFiles) {
+    if (fs.existsSync(newLocalFile)) {
+      fs.rmSync(newLocalFile, { recursive: true, force: true });
+    }
+  }
+};
+
+/**
+ * Prepare support package data: creates elastic entity and compute the S3 target folder.
+ * @param context
+ * @param user
+ * @param input
+ */
+export const prepareNewSupportPackage = async (context: AuthContext, user: AuthUser, input: SupportPackageAddInput) => {
+  const settings = await getSettings(context);
+  logSupport.warn('Platform settings are:', { settings });
```

Comment:
There is too much data on settings, I need to remove that (and soon will use telemetry data instead).